sdk/azcore/internal/exported/pipeline.go

```diff
@@ -57,21 +60,35 @@ func (tp transportPolicy) Do(req *Request) (*http.Response, error) {
 
 // NewPipeline creates a new Pipeline object from the specified Policies.
 // Not directly exported, but used as part of runtime.NewPipeline().
-func NewPipeline(transport Transporter, policies ...Policy) Pipeline {
+func NewPipeline(tr tracing.Tracer, transport Transporter, policies ...Policy) Pipeline {
 	// transport policy must always be the last in the slice
 	policies = append(policies, transportPolicy{trans: transport})
 	return Pipeline{
 		policies: policies,
+		tracer:   tr,
 	}
 }
 
+// SetTracer sets the provided [tracing.Tracer] on [Pipeline].
+// Not directly exported, but used when creating/cloning clients.
+func SetTracer(pl *Pipeline, tracer tracing.Tracer) {
+	pl.tracer = tracer
+}
+
 // Do is called for each and every HTTP request. It passes the request through all
 // the Policy objects (which can transform the Request's URL/query parameters/headers)
 // and ultimately sends the transformed HTTP request over the network.
 func (p Pipeline) Do(req *Request) (*http.Response, error) {
 	if req == nil {
 		return nil, errors.New("request cannot be nil")
 	}
+	// add the tracer to the in-flight request so any pipeline policies can access it
+	req.SetOperationValue(p.tracer)
```

Comment:
This is the core of the change. Maybe a good way to frame this is that for tracing, we always want to send the `tracing.Tracer` with the associated `Request`. Since `tracing.Tracer` is now a pipeline-internal constructor parameter, we always have access to it and can easily send it down the pipeline.